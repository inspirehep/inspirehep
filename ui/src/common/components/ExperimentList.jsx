import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import InlineList from './InlineList';

class ExperimentList extends Component {
  static renderExperiment(experiment) {
    return experiment.get('name');
  }

  render() {
    const { experiments } = this.props;
    return (
      <InlineList
        label="Experiments"
        items={experiments}
        renderItem={ExperimentList.renderExperiment}
      />
    );
  }
}

ExperimentList.propTypes = {
  experiments: PropTypes.instanceOf(List),
};

ExperimentList.defaultProps = {
  experiments: null,
};

export default ExperimentList;
