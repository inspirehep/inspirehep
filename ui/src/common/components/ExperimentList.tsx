import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import { Link } from 'react-router-dom';

import InlineList from './InlineList';
import { EXPERIMENTS } from '../routes';
import { getRecordIdFromRef } from '../utils';

class ExperimentList extends Component {
  static renderExperiment(experiment) {
    const experimentName = experiment.get('name');
    const experimentRecordId = getRecordIdFromRef(
      experiment.getIn(['record', '$ref'])
    );
    if (experimentRecordId) {
      return (
        <Link to={`${EXPERIMENTS}/${experimentRecordId}`}>
          {experimentName}
        </Link>
      );
    }
    return experimentName;
  }

  static getExperimentName(experiment) {
    return experiment.get('name');
  }

  render() {
    const { experiments } = this.props;
    return (
      <InlineList
        label="Experiments"
        items={experiments}
        extractKey={ExperimentList.getExperimentName}
        renderItem={ExperimentList.renderExperiment}
      />
    );
  }
}

ExperimentList.propTypes = {
  experiments: PropTypes.instanceOf(List),
};

ExperimentList.defaultProps = {
  experiments: null,
};

export default ExperimentList;
