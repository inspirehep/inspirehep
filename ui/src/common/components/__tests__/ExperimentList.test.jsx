import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ExperimentList from '../ExperimentList';

describe('ExperimentList', () => {
  it('renders arxiv categories', () => {
    const experiments = fromJS([
      { name: 'CERN-LHC-CMS' },
      { name: 'CERN-LHC-LHCb' },
    ]);
    const wrapper = shallow(<ExperimentList experiments={experiments} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
