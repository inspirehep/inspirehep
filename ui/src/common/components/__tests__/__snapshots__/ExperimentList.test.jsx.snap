// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentList renders arxiv categories 1`] = `
<div
  className="__InlineList__"
>
  <span>
    Experiments
    : 
  </span>
  <ul
    className="separate-items-with-comma"
  >
    <li
      key="Map { \\"name\\": \\"CERN-LHC-CMS\\" }"
    >
      CERN-LHC-CMS
    </li>
    <li
      key="Map { \\"name\\": \\"CERN-LHC-LHCb\\" }"
    >
      CERN-LHC-LHCb
    </li>
  </ul>
</div>
`;
